Sources/swift-build/versionData.swift

```diff
@@ -0,0 +1,36 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright 2016 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import POSIX
+import struct dep.Package
+import struct sys.Path
+import func libc.fclose
+
+func generateVersionData(rootDir: String, packages: [Package]) throws {
+    let dirPath = rootDir + "/.build/versionData/"
+    try mkdir(dirPath)
+    for pkg in packages {
+        try generateVersionData(dirPath, package:pkg)
+    }
+}
+
+private func generateVersionData(dirPath: String, package: Package) throws {
+
+    let filePath = Path.join(dirPath, "\(package.name).swift")
+    let file = try fopen(filePath, mode: .Write)
+    defer {
+      libc.fclose(file)
+    }
+
+    try fputs("public let url: String = \"\(package.url)\" \n", file)
+    let version = package.version
+    try fputs("public let version: (Int, Int, Int, [String], String?) = " +
```

Comment:
imo, it would be fine if it was just `public let version = (1, 0, 1)` as this is guessable but I have no way of knowing what is last two fields in `public let version = (1, 0, 1, [], "")` without seeing doc